src/kernel/mempool_persist.cpp

```diff
@@ -63,6 +63,7 @@ bool LoadMempool(CTxMemPool& pool, const fs::path& load_path, Chainstate& active
         }
         uint64_t num;
         file >> num;
+        LOCK2(cs_main, pool.cs);
```

Comment:
Afaict we currently import mempool txs from disk in the background and allow the node to make connections and sync the chain in parallel. Seems like this commit changes that by locking cs_main for the entire import duration. Not sure about the consequences.

Potentially relevant for #27460 as well.

Reply:
To recap, this disallows a wallet(or other TrimToSize caller) from trimming a parent before the child is entered into the mempool?

With debug builds, this draws out the startup time of bitcoind by ~1.5 minutes with 300MB mempool. ~30 seconds on non-debug build.

Reply:
To be clear I agree that this was not the right thing to do.

Reply:
As a datapoint, importing my mempool now (600MB used, 237k entries) seems to take about 25 minutes. Not a particularly fast machine (though not an rpi either), debug build, etc, so something of a worst-case scenario, but still.